### PR TITLE
Implement NuGet V2 Feed support

### PR DESCRIPTION
**Pull request by passuied**
Implement NuGet V2 Feed support.
- Added some unit tests. These cannot be run when signing is on at the moment.
- Needs way to up version

### CODE REVIEW

**Comment by passuied** [comment]
This PR should resolve issue: https://github.com/dotnet/ProjFileTools/issues/59

File: src/ProjectFileTools.NuGetSearch/Feeds/Web/NuGetV2ServiceFeed.cs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                    var pkgVersion = GetPropertyValue(document, el, "Version");
+
+                    if (pkgId.Equals(pkgId, StringComparison.OrdinalIgnoreCase))
+                    {
+                        results.Add(pkgVersion);
+                    }
+                }
+            }
+
+            return results.AsReadOnly();
```
[the remark this answers is not in the document]
Comment (passuied) [reply src/ProjectFileTools.NuGetSearch/Feeds/Web/NuGetV2ServiceFeed.cs]: this was literally copied from v3 :)
You're right probably not needed

**Comment by madskristensen** [comment]
I can release when @jaredpar or someone with admin rights merges it

**Comment by passuied** [comment]
@madskristensen @jaredpar Following up on this? How can we find out who is an admin on this repo?
Thanks,
Patrick

**Comment by passuied** [comment]
@madskristensen. I understand you don't have admin privileges but not sure who does and to find out. Does it mean this PR will never get merged? Please advise

**Comment by madskristensen** [comment]
@jaredpar @davkean @Pilchie do any of you know who owns this repo and can help look at the PR?

**Comment by Pilchie** [comment]
Adding @jjmew and @panopiconcentral and @jmarolf

**Comment by jmarolf** [comment]
I am not an admin for this repo either.  I assume someone in @mlorbetske 's org is?

**Comment by jaredpar** [comment]
Started an email to figure this out. Will add you all.

File: src/ProjectFileTools.NuGetSearch/Feeds/Web/NuGetV2ServiceFeed.cs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return PackageNameSearchResult.Failure;
+            }
+
+            IReadOnlyList<string> results = new List<string>();
+            string frameworkQuery = !string.IsNullOrEmpty(queryConfiguration.CompatibiltyTarget) ? $"&targetFramework={queryConfiguration.CompatibiltyTarget}" : "";
+            var serviceEndpoint = $"{_feed}/Search()";
+            Func<string, string> queryFunc = x => $"{x}?searchTerm='{prefix}'{frameworkQuery}&includePrerelease={queryConfiguration.IncludePreRelease}";
```
Comment (loic-sharma) [inline src/ProjectFileTools.NuGetSearch/Feeds/Web/NuGetV2ServiceFeed.cs]: Is this intended to exclude semver 2 packages? If not, consider something like:

```suggestion
            Func<string, string> queryFunc = x => $"{x}?searchTerm='{prefix}'{frameworkQuery}&includePrerelease={queryConfiguration.IncludePreRelease}&semVerLevel=2.0.0";
```

For example, the [`NuGet.Packaging`](https://www.nuget.org/packages/NuGet.Packaging/) package has a semver2 version `5.2.0+f6f9fb481b9fbf14154d484dbf376446c9b0d274`. The V2 search API will only return this version if explicitly include semver2 results:

* Include semver2: https://www.nuget.org/api/v2/Search()?searchTerm=nuget.packaging&semVerLevel=2.0.0
* Exclude semver2: https://www.nuget.org/api/v2/Search()?searchTerm=nuget.packaging

Comment (passuied) [reply src/ProjectFileTools.NuGetSearch/Feeds/Web/NuGetV2ServiceFeed.cs]: Makes sense! Added query string.
Thanks for the feedback!

**Comment by passuied** [comment]
@loic-sharma are you an admin on the repo?
Are you able to merge it?

**Comment by loic-sharma** [comment]
I only commented because I'm familiar with the NuGet protocol. I'm not an admin and I don't know who is, sorry 😞

**Comment by passuied** [comment]
Oh well... The search continues then... :(

**Comment by passuied** [comment]
@jaredpar any update on repo admin search?

**Comment by passuied** [comment]
@jaredpar @madskristensen it's been another 3 months. Any way to fork the repo and move the build to new repo?

**Comment by KirillOsenkov** [comment]
Turns out I have the rights to merge PRs. Apologies for the delay and thanks for the contribution!

**Comment by passuied** [comment]
Thanks! Glad we found someone with rights!

**Comment by passuied** [comment]
@KirillOsenkov @jaredpar @madskristensen who could trigger a build to release new version?

**Comment by KirillOsenkov** [comment]
@madskristensen is busy this week but next week hopefully we will find time to create a new CI that can sign the VSIX and publish. The old CI for this repo was before Azure DevOps so we'll need to create a pipeline for Azure DevOps to revive this. Thanks for your patience.

**Comment by passuied** [comment]
@madskristensen @KirillOsenkov any update on updating CI to Azure DevOps?
Please advise,
Patrick

**Pull request merged**